client/src/components/relatedItems/RelatedItems.jsx

```diff
@@ -77,75 +33,59 @@ var RelatedItems = (props) => {
 
       axios.all([productInfoRequest, productReviewRequest, productStyleRequest])
         .then(axios.spread((...responses) => {
-          outfitInfo.push(responses[0].data);
-          outfitReviews.push(responses[1].data);
-          outfitStyles.push(responses[2].data);
+          addProdInfo.push(responses[0].data);
+          addProdReviews.push(responses[1].data);
+          addProdStyles.push(responses[2].data);
         }))
         .then(() => {
           count++;
-          if (count === products.length) {
-            cb(null, outfitInfo, outfitReviews, outfitStyles);
+          if (count === products.length && count === addProdInfo.length && count === addProdReviews.length && count === addProdStyles.length) {
+            addProdInfo = [...productInfo, ...addProdInfo];
+            addProdReviews = [...productReviews, ...addProdReviews];
+            addProdStyles = [...productStyles, ...addProdStyles];
+            setProductInfo(addProdInfo);
+            setProductReviews(addProdReviews);
+            setProductStyles(addProdStyles);
           }
         })
         .catch( err => { console.log(err); });
     });
   });
 
   useEffect(() => {
-    const listType = 'current';
-    getProductInfo(productId, listType);
-  }, [productId]);
-
-  useEffect(() => {
-    const listType = 'related';
-    getRelatedProductInfo(relatedProduct, function(err, relProdInfo, relProdReviews, relProdStyles) {
-      if (err) {
-        console.log(err);
-      } else {
-        setRelatedProductInfo(relProdInfo);
-        setRelatedProductReviews(relProdReviews);
-        setRelatedProductStyles(relProdStyles);
+    let addItems = [];
+    for (var i = 0; i < allProduct.length; i++) {
+      let addItem = productInfo.filter((prod) => prod.id === allProduct[i]);
+      if (addItem.length === 0) {
+        addItems.push(allProduct[i]);
       }
-    });
-  }, [JSON.stringify(relatedProduct)]);
-
-  useEffect(() => {
-    const listType = 'outfit';
-    getYourOutfitInfo(yourOutfit, function(err, outfitInfo, outfitReviews, outfitStyles) {
-      if (err) {
-        console.log(err);
-      } else {
-        setYourOutfitInfo(outfitInfo);
-        setYourOutfitReviews(outfitReviews);
-        setYourOutfitStyles(outfitStyles);
-      }
-    });
-  }, [JSON.stringify(yourOutfit)]);
+    }
+    if (addItems.length > 0) {
+      getProductInfo(addItems);
+    }
+  }, [JSON.stringify(allProduct)]);
```

Comment:
Assuming using allProduct here to have component rerender whenever there's a change in productId, relatedProduct, or yourOutfit?
Is it because we're trying to minimize api calls? Would love to hear more about why this is done this way. We can talk on Wednesday during class.